### DISCUSSION TITLE
Webhook result also based on HTTP status code

### DISCUSSION THREAD

**Discussion by randallrecobee** [vernemq/vernemq#2240] [Ideas]
The current form of `auth_on_register` webhook can return two things, for the most part: either the connecting client is allowed or rejected. These two translate to MQTT Return Code(RC) 0 or 5, respectively. This, however, does not reflect the current state of the service behind the webhook. For instance, the service may have been overloaded, restarted, or was undergoing maintenance. All of these would result in RC 5 which depending on the client could result in different behaviors.

Would it be a good idea to take into consideration the HTTP response code of the webhook service. For example, any HTTP status code 5XX and 429 would result to RC 3(Connection Refused, Server unavailable). 

Reference:
* [MQTT Specification 3.1.1 # Connect Return code](https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349256)
* [RFC6585 - Additional HTTP Status Codes](https://datatracker.ietf.org/doc/html/rfc6585)
* [VerneMQ Webhook # auth_on_register](https://docs.vernemq.com/plugin-development/webhookplugins#auth_on_register)

**Comment by ioolkos** [comment]
@randallrecobee Thanks, I think that's a great idea. As you suggest, we need to find the minimal (and generic) mapping between HTTP and MQTT connect response codes.

Also, for consistency, we obviously want to do this for `auth_on_register_m5` too. MQTT v5 has way more Connect Reason Codes:
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901079

Can you suggest your HTTP2MQTT mappings in a Markdown table here? We can then add the v5 mappings as well.

**Reply by randallrecobee** [reply]
I would suggest something like this:
* HTTP 429 -> RC 3
* HTTP 401 -> RC 5
* HTTP 5XX -> RC 3
* HTTP 4XX -> RC 4

And, keep the original behavior intact:
* {result: OK} -> RC 0
* {result: {error: reason}} -> RC 5
